clients/cli/src/utils.rs

```diff
@@ -184,7 +193,7 @@ pub fn print_last_successful_sync() -> CliResult<()> {
     Ok(())
 }
 
-pub fn set_up_auto_save(id: Uuid, location: String) -> Option<Hotwatch> {
+pub fn set_up_auto_save(id: Uuid, location: PathBuf) -> Option<Hotwatch> {
```

Comment:
We used to accept a `String` here, as a rule of thumb, accepting `String`s in functions is a bad idea when you can instead accept an `&str`. Because you don't actually want ownership of a `String` and you force people to do `.clone()` type business needlessly so, rule of thumb with the old fn should have been `&str`.

If you read the docs about [`PathBuf`](https://doc.rust-lang.org/stable/std/path/struct.PathBuf.html) or [`Path`](https://doc.rust-lang.org/stable/std/path/struct.Path.html), you'll see that `PathBuf` is analogous to `String` (owned regular struct) and `Path` is analogous to an `&str`.

So here you really want to take a `Path`.

If you wanted to really go for excellence here, you could accept an `AsRef<Path>`, this is the common thing to do when dealing with the standard library.

Here is an example of this: https://github.com/lockbook/lockbook/blob/parth/hmdb/core/src/service/log_service.rs#L12

this allows a caller to hand in any of the following types:
+ `Path`
+ `String`
+ `&str`
+ `&String`
+ `PathBuf`
+ `&Path`
+ and much more.

Reply:
Applies to all of the below functions.

Reply:
I managed to get a bunch of functions working with Asref instead of the hardcoded types. However, this one you showed complained a lot in particular. This is what it took:
196`<P: 'static + AsRef< Path> + Send>`
200`.watch(location.as_ref().to_path_buf(), move |event: Event| match event {`

Reply:
I used AsRef in a few other functions. Let me know if you think they are good there